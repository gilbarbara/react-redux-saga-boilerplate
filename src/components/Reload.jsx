import React from 'react';
import styled from 'styled-components';

import { Button, Heading } from 'styled-minimal';

export const StyledReload = styled.div`
  button {
    pointer-events: all;
  }
`;

const Reload = () => (
  <StyledReload>
    <Heading as="h6" mb={3}>
      There's a new version of this app!
    </Heading>
    <Button variant="dark" outline size="sm" onClick={() => window.location.reload()}>
      Reload
    </Button>
  </StyledReload>
);

export default Reload;
