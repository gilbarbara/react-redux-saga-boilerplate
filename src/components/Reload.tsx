import { useCallback } from 'react';
import styled from '@emotion/styled';
import { Button, H6 } from '@gilbarbara/components';

export const ReloadWrapper = styled.div`
  button {
    pointer-events: all;
  }
`;

function Reload() {
  const handleClick = useCallback(() => {
    window.location.reload();
  }, []);

  return (
    <ReloadWrapper data-component-name="Reload">
      <H6 mb="lg">There's a new version of this app!</H6>
      <Button bg="black" invert onClick={handleClick} size="xs">
        Reload
      </Button>
    </ReloadWrapper>
  );
}

export default Reload;
