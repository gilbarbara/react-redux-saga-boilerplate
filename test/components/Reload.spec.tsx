import React from 'react';
import { fireEvent, render, screen } from 'test-utils';

import Reload from '~/components/Reload';

describe('Reload', () => {
  const { location } = window;

  beforeAll(() => {
    // @ts-ignore
    delete window.location;
    window.location = { ...location, reload: vi.fn() };
  });

  afterAll(() => {
    window.location = location;
  });

  it('should render properly', () => {
    render(<Reload />);

    expect(screen.getByTestId('Reload')).toMatchSnapshot();

    fireEvent.click(screen.getByRole('button'));

    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });
});
