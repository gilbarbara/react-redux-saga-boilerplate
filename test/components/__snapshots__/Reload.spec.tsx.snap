// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Reload > should render properly 1`] = `
<div
  class="css-15cg8s2"
  data-component-name="Reload"
>
  <h6
    class="css-hr59i5"
    data-component-name="H6"
  >
    There's a new version of this app!
  </h6>
  <button
    class="css-7rcxmt"
    data-component-name="Button"
    type="button"
  >
    Reload
  </button>
</div>
`;
